cupy/cuda/cupy_cub.cu

```diff
@@ -173,17 +196,11 @@ __host__ __device__ __forceinline__ complex<double> Min::operator()(const comple
 template <>
 __host__ __device__ __forceinline__ __half Min::operator()(const __half &a, const __half &b) const
 {
-#ifdef __CUDA_ARCH__
     // NumPy behavior: NaN is always chosen!
-    if (__hisnan(a)) {return a;}
-    else if (__hisnan(b)) {return b;}
-    else {return CUB_MIN(a, b);}
-#else
-    // TODO: avoid cast to float
-    if (isnan(__half2float(a))) {return a;}
-    else if (isnan(__half2float(b))) {return b;}
-    else {return __float2half(CUB_MIN(__half2float(a), __half2float(b)));}
-#endif
+    if (half_isnan(a)) {return a;}
+    else if (half_isnan(b)) {return b;}
+    else if (half_greater_than(a, b)) {return a;}
```

Comment:
```suggestion
    else if (half_less(a, b)) {return b;}
```

Reply:
`else if (half_less(a, b)) {return a;}` is right, isn't it?

Reply:
Yes.